src/Stack/Script.hs

```diff
@@ -152,31 +158,110 @@ scriptCmd opts go' = do
       then replaceExtension fp "exe"
       else dropExtension fp
 
-getPackagesFromModuleInfo
-  :: ModuleInfo
-  -> FilePath -- ^ script filename
+getPackagesFromImports
+  :: FilePath -- ^ script filename
   -> RIO EnvConfig (Set PackageName)
-getPackagesFromModuleInfo mi scriptFP = do
-    (pns1, mns) <- liftIO $ parseImports <$> S8.readFile scriptFP
-    pns2 <-
-        if Set.null mns
-            then return Set.empty
-            else do
-                pns <- forM (Set.toList mns) $ \mn ->
-                    case Map.lookup mn $ miModules mi of
-                        Just pns ->
-                            case Set.toList pns of
-                                [] -> assert False $ return Set.empty
-                                [pn] -> return $ Set.singleton pn
-                                pns' -> throwString $ concat
-                                    [ "Module "
-                                    , moduleNameString mn
-                                    , " appears in multiple packages: "
-                                    , unwords $ map packageNameString pns'
-                                    ]
-                        Nothing -> return Set.empty
-                return $ Set.unions pns `Set.difference` blacklist
-    return $ Set.union pns1 pns2
+getPackagesFromImports scriptFP = do
+    (pns, mns) <- liftIO $ parseImports <$> S8.readFile scriptFP
+    if Set.null mns
+        then return pns
+        else Set.union pns <$> getPackagesFromModuleNames mns
+
+getPackagesFromModuleNames
+  :: Set ModuleName
+  -> RIO EnvConfig (Set PackageName)
+getPackagesFromModuleNames mns = do
+    hash <- hashSnapshot
+    withSnapshotCache hash mapSnapshotPackageModules $ \getModulePackages -> do
+        mutableMapping <- mapMutablePackageModules
```

Comment:
The `script` command does not allow for any mutable packages, so if desired for simplicity, I think you could simply add an assertion that there are no mutable packages.

Reply:
I didn't know about that, is it documented somewhere?

Reply:
I don't think so, it's an implication of the requirement that there's no local config for `stack script`, and when I wrote that documentation, we didn't have the mutable/immutable breakdown yet.